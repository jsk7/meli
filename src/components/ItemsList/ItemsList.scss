@import 'src/styles/variables';

.ItemsList {
  min-height: 120vh;

  ul {
    top: 0;
    margin-top: 0;
    -webkit-margin-before: 0;
    -webkit-margin-after: 0;
    -webkit-padding-start: 0;
    list-style-type: none;
  }

  hr {
    width: 95%;
    border-color: rgba(0,0,0, 0.05);
    margin: 0 auto;
  }

  .item {
    padding: $base-margin 0;
  }
}
