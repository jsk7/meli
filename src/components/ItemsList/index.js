import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '../ListItem';
import { Link } from 'react-router';
import './ItemsList.scss';

export default class ItemsList extends React.PureComponent {

  render() {
    let { items } = this.props;

    console.log(items);
    if(!items.length) { // Placeholders
      for(let i = 0; i < 4; i++)
        items = [...items, {
          id: 'placeholder-' + i,
          title: "",
          price: "",
          condition: "",
          seller_address: {state: {}}
        }];
    }

    return (
      <div className="ItemsList">
        <ul>
          {
            items.map(item =>
              <li key={item.id}>
                <div className="item">
                  <Link to={`/items/${item.id}`}>
                    <ListItem item={item}
                                    lang={this.props.lang}
                    />
                  </Link>
                </div>
                <hr />
              </li>
            )
          }
        </ul>
      </div>
    )
  }
}

ItemsList.propTypes = {
  items: PropTypes.array,
  lang: PropTypes.object.isRequired
}
