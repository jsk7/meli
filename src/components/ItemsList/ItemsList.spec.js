import React from 'react';
import ItemsList from './index';
import renderer from 'react-test-renderer';
import itemsMock from '../../mocks/items';


describe('<ItemsList />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ItemsList items={itemsMock} lang={{}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});



