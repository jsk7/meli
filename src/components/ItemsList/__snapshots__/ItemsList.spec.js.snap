// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsList /> should render correctly 1`] = `
<div
  className="ItemsList"
>
  <ul>
    <li>
      <div
        className="row"
      >
        <div
          className="item"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="ListItem"
            >
              <div
                className="image-helper"
              >
                <img
                  className="ItemQuickImage"
                  src="http://mla-s2-p.mlstatic.com/893911-MLA25840454871_082017-I.jpg"
                />
              </div>
              <div
                className="description col-9 col-7-sm"
              >
                <div
                  className="header"
                >
                  <h2
                    className="price"
                  >
                    $ 7500
                  </h2>
                  <img
                    className="shipping"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <h4
                    className="location"
                  >
                    Capital Federal
                  </h4>
                </div>
                <h3
                  className="title"
                >
                  <span
                    className="capitalize"
                  >
                    Apple Ipod Touch 32gb 6ta Generacion
                  </span>
                </h3>
                <h3
                  className="title"
                >
                  <span
                    className="capitalize"
                  />
                </h3>
              </div>
            </div>
          </a>
        </div>
        <hr />
      </div>
    </li>
    <li>
      <div
        className="row"
      >
        <div
          className="item"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="ListItem"
            >
              <div
                className="image-helper"
              >
                <img
                  className="ItemQuickImage"
                  src="http://mla-s1-p.mlstatic.com/636945-MLA25667589604_062017-I.jpg"
                />
              </div>
              <div
                className="description col-9 col-7-sm"
              >
                <div
                  className="header"
                >
                  <h2
                    className="price"
                  >
                    $ 1700
                  </h2>
                  <img
                    className="shipping"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <h4
                    className="location"
                  >
                    Buenos Aires
                  </h4>
                </div>
                <h3
                  className="title"
                >
                  <span
                    className="capitalize"
                  >
                    Ipod Shuffle 2gb Apple 5ta Generacion Pink / Blue Rosa Azul
                  </span>
                </h3>
                <h3
                  className="title"
                >
                  <span
                    className="capitalize"
                  />
                </h3>
              </div>
            </div>
          </a>
        </div>
        <hr />
      </div>
    </li>
  </ul>
  <div
    className="sampleContainer"
  >
    <div
      className="loader"
    >
      <span
        className="dot dot_1"
      />
      <span
        className="dot dot_2"
      />
      <span
        className="dot dot_1"
      />
      <span
        className="dot dot_4"
      />
    </div>
  </div>
</div>
`;
